### PR TITLE
STM32: Convert I2S to devicetree pinctrl configuration

### PR DESCRIPTION
**Pull request by erwango**
Last STM32 driver to be converted to  device tree based pinctrl configuration

### CODE REVIEW

**Comment by erwango** [comment]
@avisconti Mostly relying on you to test this :-)

**Review by galak — APPROVED** [review]

File: dts/bindings/i2s/st,stm32-i2s.yaml [hunk 1]
```diff
@@ -19,3 +19,12 @@ properties:
...
       required: true
+
+    pinctrl-0:
+      type: phandles
+      required: false
+      description: |
+        Pin configuration for I2S signals.
+        We expect that the phandles will reference pinctrl nodes.
+
+        For example the I2S1 would be
```
Comment (galak) [inline dts/bindings/i2s/st,stm32-i2s.yaml:30]: I think you forgot something here.

**Review by galak — CHANGES_REQUESTED** [review]
binding looks like its missing or got cut-off.

**Review by galak — COMMENTED** [review]
Can you add a comment in `boards: stm32: Convert I2S users to dt based pinctrl config` as to what was going on with nucleoo_f411re (since the block removed has SPI signal names)

**Review by galak — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

**Pull request merged**